### PR TITLE
Explosion component, plus some other spell part things

### LINKED ISSUE #284: Explosion Component

### PR DESCRIPTION
**Pull request by IchHabeHunger54**

### CODE REVIEW

File: src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/spell/IPrefabSpellManager.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public interface IPrefabSpellManager extends Map<ResourceLocation, IPrefabSpell> {
+    IPrefabSpell getOrDefault(@Nullable ResourceLocation id, Supplier<IPrefabSpell> defaultSupplier);
+
+    Optional<IPrefabSpell> getOptional(@Nullable ResourceLocation id);
+
+    Map<ResourceLocation, IPrefabSpell> getData();
+}
```
Comment (Minecraftschurli) [inline src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/spell/IPrefabSpellManager.java]: documentation needed

File: src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/spell/ISpellTransformationManager.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.github.minecraftschurlimods.arsmagicalegacy.api.spell;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+import java.util.Optional;
+
+/**
+ *
```
Comment (Minecraftschurli) [inline src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/spell/ISpellTransformationManager.java]: documentation needed

File: src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/init/AMSpellParts.java [hunk 3]
```diff
@@ -43,7 +43,6 @@ public interface AMSpellParts {
     RegistryObject<Wall>               WALL                = SPELL_PARTS.register("wall",                Wall::new);
     RegistryObject<Wave>               WAVE                = SPELL_PARTS.register("wave",                Wave::new);
     RegistryObject<Zone>               ZONE                = SPELL_PARTS.register("zone",                Zone::new);
-    // TODO contingencies: damage, death, fall, fire, health
```
Comment (Minecraftschurli) [inline src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/init/AMSpellParts.java:46]: leave this in

**Review by Minecraftschurli — CHANGES_REQUESTED** [review]
Some little things

File: src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/init/AMSpellParts.java (hunk 3 above)
Comment (IchHabeHunger54) [reply src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/init/AMSpellParts.java:46]: Why? It's been added to the project board, should be fine to remove this.

**Review by Minecraftschurli — APPROVED** [review]
Looks good to me (I hope you tested everything client and server) :shipit:

**Pull request merged**